snakes_cafe.py

```diff
@@ -72,71 +77,133 @@ def ordering():
-                            sub_total()
-                            print('{} {}'.format('Your current total is now: ',
-                                  '$' + str(sub_total())))
+                print('Your entered quantity is not valid.')
+                continue
+        else:
+            user_input_item = user_input
+            user_input_qty = 1
+        if user_input_item == 'Quit':
+            condition = False
+            break
+        elif user_input_item == 'Order':
+            print_receipt()
+        elif user_input_item == 'Remove':
+            print('Enter name of item that you want to remove')
+            which_item_remove = input('>' + '\t')
+            which_item_remove = which_item_remove.title()
+            for key in cart.keys():
+                if which_item_remove in key:
+                    del cart[key]
+                    continue
+            print('The item is not in your order.')
+            continue
+        elif user_input_item == 'Menu':
+            menu_items()
+        elif user_input_item in menus.keys():
+            for key, value in menus.items():
+                if user_input_item == key:
+                    for key in value.keys():
+                        print(key[0])
+        else:
+            if not item_check(user_input_item):
+                continue
+            adding_item_to_cart(user_input_item, user_input_qty)
+            print('** {} order of {} have been added'
+                  .format(user_input_qty, user_input_item))
+            print('Your current total is now: ${}'
+                  .format(str(sub_total())))
+
+
+# Check if item is in menu
+def item_check(item_name):
+    for value in menus.values():
+        if item_name in [key[0] for key in value]:
+            return True
+    print("We don't have that item.")
+    return False
+
+
+# check if we have in qty the amnt being ordered
+def quantity_check(num):
+    global user_input
+    user_input = user_input.title()
+    item_name = user_input.split(', ')[0]
+    for value in menus.values():
+        for tuple_item in value:
+            if item_name == tuple_item[0]:
+                if num > value[tuple_item]:
+                    print("We don't have that many.")
+                    return False
+                return True
+    return False
+
+
+def adding_item_to_cart(item_name, num_of_item=1):
+    """ 
+    adds item to cart after use orders it
+    """
+    prev_num = 0
+    for key, value in menus.items():
+        for tuple_item in value:
+            if item_name == tuple_item[0]:
+                if tuple_item[0] in cart:
+                    prev_num = cart[tuple_item[0]][tuple_item[1]]
+                    cart[tuple_item[0]].update({tuple_item[1]:
+                                                prev_num + num_of_item})
+                    break
+                else:
+                    cart[tuple_item[0]] = {tuple_item[1]: num_of_item}
+                    break
```

Comment:
You are not removing this item from stock as well as adding it to the cart. Thus, the user can keep adding items even if they end up with more of an item than available in stock.